tensorflow/python/ops/array_ops.py

```diff
@@ -3177,6 +3181,46 @@ def where(condition, x=None, y=None, name=None):
     raise ValueError("x and y must both be non-None or both be None.")
 
 
+@tf_export("where", v1=["where_v2"])
+def where_v2(condition, x=None, y=None, name=None):
```

Comment:
Is there ever a set of input shapes for which where and where_v2 both give answers but the answers are different?

I assume not, unless you're doing broadcast wrong.

In this case instead of a new API function you should go through forward compatibility, using the [compat.forward_compatible](https://github.com/tensorflow/tensorflow/blob/bd92ce08f8d5ae2dfa5cf47f39d86710215b7185/tensorflow/python/compat/compat.py#L34) function set to about a month in the future from now.

Search the tensorflow/python/ops directory for other uses of compat to see how to do this.

Reply:
Ah I see.

Then you also need to add a rule to [tf_upgrade_v2](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/compatibility/tf_upgrade_v2.py) to convert tf.where_v2 to tf.compat.v2.where as otherwise there can be a confusion when someone runs the upgrade script and tries to use their code in both v1 and v2.

Reply:
Since I wrote this message the tf v2 API has been frozen for the 1.14 release. This means we'll need to export this symbol as where_v2 both in tf1 and in tf2 :-/